app/scripts/modules/core/application/modal/applicationProviderFields.directive.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const angular = require('angular');
+
+module.exports = angular
+  .module('spinnaker.core.application.modal.applicationProviderFields.directive', [
+    require('../../cloudProvider/cloudProvider.registry'),
+    require('../../utils/lodash.js'),
+  ])
+  .directive('applicationProviderFields', function() {
```

Comment:
It's preferable to use components when possible. This would become:
```javascript
.component('applicationProviderFields', {
  bindings: {
    application: '=',
    cloudProviders: '='
  },
  templateUrl: require('./applicationProviderFields.component.html')
})
```
Then, replace `vm` with `$ctrl` in the template.